src/ReleaseApp/Infrastructure/Client/HttpRequestExecutor.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace ReleaseApp\Infrastructure\Client;
+
+use GuzzleHttp\Client as GuzzleHttp;
+use GuzzleHttp\Exception\ServerException;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use ReleaseApp\Infrastructure\Configuration\ConfigurationProvider;
+use Upgrade\Application\Exception\UpgraderException;
+
+class HttpRequestExecutor implements HttpRequestExecutorInterface
+{
+    /**
+     * @var \GuzzleHttp\Client
+     */
+    protected $guzzleClient;
+
+    /**
+     * @var \ReleaseApp\Infrastructure\Configuration\ConfigurationProvider
+     */
+    protected ConfigurationProvider $config;
+
+    /**
+     * @param \ReleaseApp\Infrastructure\Configuration\ConfigurationProvider $config
+     */
+    public function __construct(ConfigurationProvider $config)
+    {
+        $this->guzzleClient = new GuzzleHttp();
+
+        $this->config = $config;
+    }
+
+    /**
+     * @param \Psr\Http\Message\RequestInterface $request
+     *
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     *
+     * @return \Psr\Http\Message\ResponseInterface
+     */
+    public function requestExecute(RequestInterface $request): ResponseInterface
+    {
+        $attempts = 0;
+        $exception = null;
+        $guzzleResponse = null;
+
+        do {
+            try {
+                $guzzleResponse = $this->guzzleClient->send($request);
+            } catch (ServerException $currentException) {
+                $exception = $currentException;
+                sleep($this->config->getHttpRetrieveRetryDelay());
+            } finally {
+                ++$attempts;
```

Comment:
Why it's necessary?

Reply:
For counting how many requests was send

Reply:
Upgrade makes 3 attempts to get a response from ReleaseAPP, then return error, if no correct result was received. 
We need to count amount  of attempts